node/src/components/consensus.rs

```diff
@@ -260,7 +260,16 @@ where
                     panic!("couldn't get the seed from the key block");
                 });
                 let validators = match get_validators_result {
-                    Ok(Some(result)) => result,
+                    Ok(Some(validator_weights)) => validator_weights
+                        .into_iter()
+                        .filter_map(|(key, stake)| match key.try_into() {
+                            Ok(key) => Some((key, stake)),
+                            Err(error) => {
+                                error!(%error, "error converting the bonded key");
```

Comment:
Should we panic here? It would mean that contract runtime (auction contract) returns incompatible data. That would be a software bug most probably.

Reply:
It used to be `warn!`, and I changed it to `error!`… however, I'm rather thinking I should change it back to `warn!`: Anyone can bond with an invalid key, can't they? So the system shouldn't crash because of that.

Reply:
I think EE shouldn't allow for that. What if _everyone_ bonds with an invalid key? Will we get era with no validators?

Reply:
It looks like it's possible (see discussion on Slack).
We would panic [here](https://github.com/CasperLabs/casper-node/blob/e8188f12d86f4efed6baa601a3fda04e20ebdcf1/node/src/components/consensus/protocols/highway.rs#L79) if that happened.

Reply:
I think EE should be verifying the public key on its end and we should panic here. It feels wrong to allow the public key into an auction, win the slot and then fail.

Reply:
But as long as that's not happening, it would be dangerous to panic here. That means anyone could make the network crash.

Reply:
I think EE should fix it then.